Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CXFSAMLCallbackHandler.java

```diff
@@ -92,7 +94,10 @@ public void handle(final Callback[] callbacks) throws IOException, UnsupportedCa
                     final Message message = getCurrentMessage();
                     AssertionType custAssertion = getCustAssertion(message.get("assertion"));
                     NhinTargetSystemType systemType = (NhinTargetSystemType) message.get("targetSystemType");
-                    String exchangeName = systemType != null ? systemType.getExchangeName() : StoreUtil.INTERNAL_EXCHANGE;
+                    String exchangeName = systemType != null ? systemType.getExchangeName() : InternalExchangeManager
```

Comment:
Why we are using InternalExchangeManager here? We have defaultExchange in exchangeInfo as well. Are they going to be the same thing. If yes we need to document this and possibly update regression suites

Reply:
We handle this in case for secure adapter scenario